### PR TITLE
copr: Add vectorized `interval_real`

### PR DESCRIPTION
**Pull request by fky2015**
Signed-off-by: FKYnJYQ <loveress01@gmail.com>

PCP #5751 

###  What have you changed?

Add vectorized `interval_real` 

###  What is the type of the changes?

- Improvement

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by fky2015** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/116

**Comment by TennyZhuang** [comment]
It seems that multiple `interval_*` function can be done with generics?

**Comment by fky2015** [comment]
I would like to when I first see these two functions `interval_int` and `interval_real`. But when I looked into `interval_int`,  it seems that this function presume that every parameter passed into `interval` may either be `u64` or `i64`, as you can see below:

https://github.com/tikv/tikv/blob/281b7d440708428f23f808a3aae76b18d4eec3ad/components/tidb_query/src/expr/builtin_compare.rs#L292-L322

So I'm not sure if it's feasible to use generics, or may be we could just leave behind this presumption.

**Comment by AndreMouche** [comment]
@andylokandy @breeswish PTAL and the next deadline for reviewing is 2020-02-20 24:00:00

File: components/tidb_query/src/rpn_expr/impl_compare.rs [hunk 1]
```diff
@@ -227,6 +227,29 @@ pub fn coalesce<T: Evaluable>(args: &[&Option<T>]) -> Result<Option<T>> {
...
 
+#[rpn_fn(varg)]
+#[inline]
+pub fn interval_real(args: &[&Option<Real>]) -> Result<Option<Int>> {
+    let target = match args[0] {
+        None => return Ok(Some(-1)),
+        Some(v) => *v,
+    };
+
+    let mut left = 1;
```
Comment (sticnarf) [inline components/tidb_query/src/rpn_expr/impl_compare.rs]: Would you mind using [slice::binary_search_by](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.binary_search_by) or [slice::binary_search_by_key](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.binary_search_by_key)?

**Pull request closed without merging**